pkg/ip/link.go

```diff
@@ -66,27 +73,76 @@ func makeVeth(name string, mtu int) (peerName string, veth netlink.Link, err err
 	return
 }
 
-// RandomVethName returns string "veth" with random prefix (hashed from entropy)
-func RandomVethName() (string, error) {
-	entropy := make([]byte, 4)
+const ENTROPY_LEN = 4
+
+// RandomIfaceName returns string @prefix with random suffix (hashed from entropy)
+func RandomIfaceName(prefix string) (string, error) {
+	if len(prefix)+ENTROPY_LEN > syscall.IFNAMSIZ {
+		return "", fmt.Errorf("prefix %s too long", len(prefix))
```

Comment:
if we can't predict the maximum prefix length statically and document it, we should at least give it in the error message.